### ISSUE TITLE
[Feature] peripheral.wrap for Immersive Engineering's Redstone Wire Connector

### ISSUE THREAD

**Issue by Jason-c-o** [Seniorendi/AdvancedPeripherals#29]
**Describe your feature idea**
Allow the Redstone Wire Connector from Immersive Engineering to be used as a peripheral, with an API to control it's input/output status and its redstone colour.

**Additional context**
https://ftb.gamepedia.com/Redstone_Wire_Connector_(Immersive_Engineering)

For some reason, CC:Tweaked does not properly read the colour of IE's redstone signals through it's redstone wires and connectors. (getBundledInput() does not return a colour). This would allow a computer to still send a coloured redstone signal in IE by changing the colour of the connector before sending a signal.

I've tried talking to the IE mod team, and they don't seem interested in finding out why it's not working properly. This is a bit of an indirect, but functional, fix. Not really your problem, but would be massively appreciated!

**Comment by Seniorendi** [comment]
Sounds good and useful, I will implement this in the future.

**Comment by Kollock5** [comment]
new release of IE already adds that feature

> - First 1.16.5 release
> - Allowed CC:Tweaked computers to emit bundled redstone signals using interface connectors (Malte)

[here](https://github.com/BluSunrize/ImmersiveEngineering/commit/12c78f73934473c1f28dda84f3ccf6db05d3cde6)

can be closed

**Comment by Jason-c-o** [comment]
> 
> 
> new release of IE already adds that feature
> 
> > * First 1.16.5 release
> > * Allowed CC:Tweaked computers to emit bundled redstone signals using interface connectors (Malte)
> 
> [here](https://github.com/BluSunrize/ImmersiveEngineering/commit/12c78f73934473c1f28dda84f3ccf6db05d3cde6)
> 
> can be closed

Sort of. It allows the Redstone Interface Connector to read an output from a computer, but input is still not implemented. I was playing around with this the other day to see if it was worth deleting this suggestion.

It would still be kind of cool to be able to alter the configuration of attached Redstone Connectors as peripherals.

**Comment by Seniorendi** [comment]
Hey, i want to add this feature in 0.5b. What functions do you want?
Something like:
```
getChannel()
getSignalForChannel(int color)
setChannel(int color)
```

**Comment by Jason-c-o** [comment]
Yeah, those sound good! If I can think of something else I'll comment!

**Comment by Seniorendi** [comment]
Hey, I have a lil' question.
I currently work on an Immersive Engineering integration but I saw https://github.com/BluSunrize/ImmersiveEngineering/commit/17e8a953412d08a7baefa7926b2297ed6ec62bbe

Does IE already have cc integration?

**Comment by Jason-c-o** [comment]
They didn't when I made the request.

I think now they've added some compatibility. There's not a lot to it though.

I haven't played in a while now so that also may have changed.

**Comment by Seniorendi** [comment]
Fixed in 0.6r
